edgelet/iotedged/src/lib.rs

```diff
@@ -759,6 +1003,104 @@ fn manual_provision(
     tokio_runtime.block_on(provision)
 }
 
+#[allow(clippy::too_many_arguments)]
+fn dps_x509_provision<HC, M>(
```

Comment:
It seems like a bunch of this should live in one of `edgelet-` crates or maybe `provisioning`? It feels like there is too much stuff happening in effectively `main`.

Reply:
Agreed. Will add a task to the backlog.